### PR TITLE
support faster route events in NavigationView and supporting classes

### PR DESCRIPTION
**Pull request by cafesilencio**
#2894 Description
Make the UI SDK compatible with faster route observer

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

### Goal

Added support for faster routes in the UI SDK specifically the NavigationView and supporting classes.

### Implementation

Added a faster route observer to the NavigationViewModel and when the route is faster the route is updated in the core.

## Screenshots or Gifs

N/A

## Testing

- [ ] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [ ] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationViewModel.java [hunk 1]
```diff
@@ -313,7 +316,7 @@ void sendEventFailedReroute(String errorMessage) {
     return isFeedbackSentSuccess;
   }
 
-  LiveData<Location> retrieveLocationUdpates() {
+  LiveData<Location> retrieveLocationUpdates() {
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationViewModel.java:319]: Nice catch!

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationViewModel.java [hunk 2]
```diff
@@ -63,11 +65,11 @@
...
-  public final MutableLiveData<InstructionModel> instructionModel = new MutableLiveData<>();
-  public final MutableLiveData<BannerInstructionModel> bannerInstructionModel = new MutableLiveData<>();
-  public final MutableLiveData<SummaryModel> summaryModel = new MutableLiveData<>();
-  public final MutableLiveData<Boolean> isOffRoute = new MutableLiveData<>();
-  public final MutableLiveData<RouteJunctionModel> routeJunctionModel = new MutableLiveData<>();
+  private final MutableLiveData<InstructionModel> instructionModel = new MutableLiveData<>();
+  private final MutableLiveData<BannerInstructionModel> bannerInstructionModel = new MutableLiveData<>();
+  private final MutableLiveData<SummaryModel> summaryModel = new MutableLiveData<>();
+  private final MutableLiveData<Boolean> isOffRoute = new MutableLiveData<>();
+  private final MutableLiveData<RouteJunctionModel> routeJunctionModel = new MutableLiveData<>();
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationViewModel.java:72]: For documentation purposes noting that this breaks SemVer cc @langsmith

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationViewModel.java [hunk 3]
```diff
@@ -473,13 +480,34 @@ public void onNewBannerInstructions(@NotNull BannerInstructions bannerInstructio
...
+    ) {
+      if (isAlternativeFaster) {
+        updateRoute(alternativeRoute);
+      }
     }
-  };*/
+
+    @Override
+    public long restartAfterMillis() {
+      return TimeUnit.MINUTES.toMillis(1);
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationViewModel.java:508]: Let's default to 2 minutes as in Core 👀 https://github.com/mapbox/mapbox-navigation-android/blob/6f5359fad0a0385c7213cc1e23537d1c5093e11a/libnavigation-core/src/main/java/com/mapbox/navigation/core/fasterroute/FasterRouteObserver.kt#L14

This wouldn't be necessary if we force Java 8 and add default interface method support from the Core ¯\\\_(ツ)\_/¯

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationViewModel.java (hunk 2 above)
Comment (cafesilencio) [reply libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationViewModel.java:72]: Does this really break anything? These aren't normally external developer facing are they?

Comment (Guardiola31337) [reply libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationViewModel.java:72]: You're right, although they were `public` and `NavigationViewModel` too (it's not under an `internal` package either) so technically speaking yes, it's SemVer. As discussed we should work on https://github.com/mapbox/mapbox-navigation-android/issues/2864 so we don't need to worry about this anymore.

**Comment by cafesilencio** [comment]
Faked the routes a bit but was able to demonstrate the behavior when a faster route is received.

**Pull request merged**